From 7b606ec0b8b0ffebf30b8478a16e812798039cc6 Mon Sep 17 00:00:00 2001
From: Christopher Jordan-Denny <jordan.denny5@gmail.com>
Date: Fri, 16 Aug 2019 00:41:54 -0500
Subject: [PATCH] Handle SIGHUP for daemon reloads.

---
 gpspipe.c | 21 +++++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/gpspipe.c b/gpspipe.c
index 72eed30..abf2a9d 100644
--- a/gpspipe.c
+++ b/gpspipe.c
@@ -60,6 +60,9 @@
 #include "gpsdclient.h"
 #include "revision.h"
 
+static char *outfile = NULL;
+static FILE *fp;
+
 static struct gps_data_t gpsdata;
 static void spinner(unsigned int, unsigned int);
 
@@ -134,6 +137,19 @@ static void usage(void)
 		  "You must use -o if you use -d.\n");
 }
 
+void handle_sighup(int sig)
+{
+    int ret = -1;
+    if(SIGHUP == sig)
+    {
+        fp = freopen(outfile, "w", fp);
+        if(NULL == fp)
+        {
+            exit(EXIT_FAILURE);
+        }
+    }
+}
+
 int main(int argc, char **argv)
 {
     char buf[4096];
@@ -151,13 +167,11 @@ int main(int argc, char **argv)
     long count = -1;
     int option;
     unsigned int vflag = 0, l = 0;
-    FILE *fp;
     unsigned int flags;
     fd_set fds;
 
     struct fixsource_t source;
     char *serialport = NULL;
-    char *outfile = NULL;
 
     flags = WATCH_ENABLE;
     while ((option = getopt(argc, argv, "?dD:lhrRwStT:vVn:s:o:pPu2")) != -1) {
@@ -289,6 +303,9 @@ int main(int argc, char **argv)
 	}
     }
 
+    /* Handle the signal SIGHUP. This way we can reload the daemon. */
+    signal(SIGHUP, handle_sighup);
+
     /* Open the serial port and set it up. */
     if (serialport)
 	open_serial(serialport);
-- 
2.20.1

